vm/src/stdlib/os.rs

```diff
@@ -3115,6 +3115,27 @@ mod posix {
         Ok(fd)
     }
 
+    extern "C" {
+        fn ctermid(s: *mut libc::c_char) -> *mut libc::c_char;
+    }
+    #[pyfunction]
+    fn _ctermid(vm: &VirtualMachine) -> PyResult {
+        let returned = unsafe { ctermid(std::ptr::null_mut()) };
+        if returned.is_null() {
+            Err(errno_err(vm))
+        } else {
+            let returned = unsafe { ffi::CStr::from_ptr(returned) }.to_str().unwrap();
+            Ok(vm.ctx.new_str(returned))
+        }
+    }
+
+    #[pyfunction]
+    // fchdir is the same as os.chdir(file descriptor)
+    // also check out: https://docs.rs/libc/0.2.86/libc/fn.fchdir.html
+    fn fchchdir(path: PyPathLike, vm: &VirtualMachine) -> PyResult<()> {
```

Comment:
comment says the function name is `fchdir`, but the written name is `fchchdir`.
By Python document, the function should take fd instead of PathLike.

Reply:
i will update the branch with the changes :) thanks again for the review

Reply:
quick question: i do agree that the `fd` must be passed into the function rather than the pathlike above, so should i use an extern as i did with`ctermid` ?  or is the latest version ok?

Reply:
I don't know if `env::set_current_dir` take `fd` as its argument. If it does, it will be ok. If there is any test about this funciton in `test_os.py` or `test_poisix.py`, make them run to test if it works correctly. Otherwise adding a test in `extra_tests/snippets` will be helpful. If the new functions are not fit for the snippet testing, creating a comment how you tested your new functions will be helpful.

Reply:
ill add the test cases as well - in this way i'll also get to learn how the testing is done :)

Reply:
update: when i was running the `test_os.py` file, the existing test case seems to pass:
 ```bash
test_fchdir (__main__.TestInvalidFD) ... ok
```
so i am guessing i dont need to add anything here ?